### PR TITLE
Handle labels in Bazel 6

### PR DESCRIPTION
**Pull request by jnowjack-lucidchart**
When compiling a project that's using Bazel 6, the Deps Check would fail with `java.util.NoSuchElementException: key not found: //path/to/target` because the label is now `@//path/to/target` in the `labelToPaths` map

### CODE REVIEW

**Review by BrianXDong — COMMENTED** [review]
Looks like this change should be backwards incompatible, is that accurate, and, if so, has it been tested?

**Comment by jnowjack-lucidchart** [comment]
> Looks like this change should be backwards incompatible, is that accurate, and, if so, has it been tested?

It should be backwards compatible. I still access the map first by the existing way and fallback to the new way (with the extra `@`) if that value doesn't exist. It's the `getOrElse` in `labelToPaths.getOrElse(label, labelToPaths(s"@$label")` doing that. I can test it too

**Comment by BrianXDong** [comment]
> > Looks like this change should be backwards incompatible, is that accurate, and, if so, has it been tested?
> 
> It should be backwards compatible. I still access the map first by the existing way and fallback to the new way (with the extra `@`) if that value doesn't exist. It's the `getOrElse` in `labelToPaths.getOrElse(label, labelToPaths(s"@$label")` doing that. I can test it too

Wow, I meant to say "compatible" not "incompatible." Though, the point about testing it still stands.

**Comment by jnowjack-lucidchart** [comment]
>
> Wow, I meant to say "compatible" not "incompatible." Though, the point about testing it still stands.

Haha nice. Testing on Bazel 5 worked just fine, so it's good for both.

**Review by BrianXDong — APPROVED** [review]

**Comment by jjudd** [comment]
Can we please update the PR to change the Bazel versions to 6 and 5 instead of 5 and 4? https://github.com/lucidsoftware/rules_scala/blob/lucid-master/tools/bazel#L39

**Comment by jjudd** [comment]
This is scope creep on this PR, but if we don't care about the macos builds anymore, then we should just disable them.

File: tests/dependencies/indirect/test [hunk 1]
```diff
@@ -3,15 +3,15 @@
 
 # used_deps off
 ! bazel build :used_deps_off_c || false
-bazel build :used_deps_off_c |& grep "buildozer 'add deps //dependencies/indirect:used_deps_off_a' //dependencies/indirect:used_deps_off_c"
+bazel build :used_deps_off_c |& grep "buildozer 'add deps @\?//dependencies/indirect:used_deps_off_a' //dependencies/indirect:used_deps_off_c"
```
Comment (jnowjack-lucidchart) [inline tests/dependencies/indirect/test:6]: In Bazel 6 the `@` exists while it doesn't in Bazel 5, therefore the optional check here. I have verified that buildozer handles the `@` correctly and will not add it to the `BUILD.bazel` file (it adds the dep without the `@`)

**Review by jjudd — APPROVED** [review]

**Pull request merged**